internal/ctxutil/ctx.go

```diff
@@ -0,0 +1,38 @@
+package ctxutil
+
+import (
+	"context"
+	"github.com/TykTechnologies/tyk/internal/proxy"
+	"net/http"
+)
+
+const (
+	upstreamAuth = "upstream-auth"
```

Comment:
Can you use a context struct for the key, so we make this harder to access?

```
type contextKey string
upstreamAuthKey = contextKey("upstream-auth")
```

either that, or add it into the ctx package iota.